### PR TITLE
269 task add storage of qc aqms and check results in a dataframe

### PR DESCRIPTION
**Pull request by ibevers**
## Description

- Stores outputs of AQMs (`metrics`) and `checks` in a data frame. 
- Uses a unified `apply_audio_quality_function` to run each check and metric function and update the dataframe. 
- Removes storage of classified results in an updated taxonomy tree, meaning that the taxonomy tree is no longer modified.
- Adds a metrics field to each class in the taxonomy tree.
- Updates the taxonomy tree to store empty metrics and checks as empty lists rather than `None`.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/senselab/audio/tasks/bioacoustic_qc/checks.py [hunk 1]
```diff
@@ -13,64 +13,19 @@
...
 import torch
 
 from senselab.audio.data_structures import Audio
 
 
-def audio_length_positive_check(audios: List[Audio], activity_audios: List[Audio]) -> Dict:
-    """Checks if an Audio object has a positive length.
+def audio_length_positive_check(audio: Audio) -> bool:
+    """Checks if an Audio object has a positive length."""
+    return audio.waveform.numel() != 0
```
Comment (fabiocat93) [inline src/senselab/audio/tasks/bioacoustic_qc/checks.py:26]: here and later in this same script, you could reuse existing features and avoid duplicate code

File: src/senselab/audio/tasks/bioacoustic_qc/constants.py [hunk 2]
```diff
@@ -5,68 +5,79 @@
 BIOACOUSTIC_ACTIVITY_TAXONOMY = {
```
Comment (fabiocat93) [inline src/senselab/audio/tasks/bioacoustic_qc/constants.py:5]: I have the feeling that there is a recursive schema in your `BIOACOUSTIC_ACTIVITY_TAXONOMY` that would be more easily maintainable with a recursive class structure

**Review by fabiocat93 — COMMENTED** [review]
Hi @ibevers, sorry for the delay in reviewing your PR (I was assisting with other branches).

## Development approach + unasked suggestions
If I were in your shoes, I’d structure the development process as follows:

1. Integrate a decision maker first (the core engine that determines whether an audio sample is noisy or not) and allow users to define their own rules, enabling them to experiment with different configurations and evaluate the feasibility of a rule-based QA framework.
2. Implement and validate individual rules. Once the framework is functional, introduce predefined rules (your duration and energy checks are good basic examples).
3. Organize and structure rules – After validating the approach, categorize rules by task and relevant conditions to ensure clarity and usability. This way, you will provide your users with some validated templates and instructions on how to use the rules that we internally test.

Currently, the development seems to be following the reverse approach—starting with a taxonomy of tasks and rules before testing their practical effectiveness. This could be risky, as it might lead to unnecessary implementation efforts. However, I assume you’ve aligned with @satra on the conceptual framework and development priorities, so I’ll limit my input to leaving comments on the code >>>>>>>>>>>>>>

## Code feedback
Generally, I don't see any major red flags, but I noticed that some functions may be reinventing existing solutions. For example, Snorkel’s `LFApplier` already provides functionality to apply multiple checks, output results in different formats, and even include a progress bar. If we’re considering using Snorkel’s engine, reusing their utilities might be a more efficient approach rather than implementing custom solutions. I’m unsure whether they support parallelization, but neither do we at the moment. What's the advantage of your new implementation then? I have left some more comments and suggestions here and there.

 Cheers 🚀

**Comment by ibevers** [comment]
Thanks for your input @Fabio and patience with my reply (lots to consider above). 

> Integrate a decision maker first

I've implemented AQMs since we want to include those in the upcoming B2AI data release. The framework here is meant to derive AQMs and check results in a decision-making agnostic way since decision-making with a model will be dataset specific for now. Decision-making might be integrated into the structure of the taxonomy through automatic exclusion. Evaluating a LabelModel requires a labeled test set, and the framework for deriving relevant AQMs and checks will facilitate creating such test sets. Also, multiple label functions may use the same AQMs, so I'm prioritizing AQMs over label functions/checks. 

The current taxonomy is provisional and will be refined (https://github.com/sensein/senselab/issues/298). 

That's a good point about using the LFApplier. I plan to explore integrating that.

> I’m unsure whether they support parallelization, but neither do we at the moment.

When I get to parallelization (https://github.com/sensein/senselab/issues/297), I'll consider whether LFApplier is the best approach.

File: src/senselab/audio/tasks/bioacoustic_qc/constants.py (hunk 2 above)
Comment (ibevers) [reply src/senselab/audio/tasks/bioacoustic_qc/constants.py:5]: Good point: https://github.com/sensein/senselab/issues/296

File: src/senselab/audio/tasks/bioacoustic_qc/checks.py (hunk 1 above)
Comment (ibevers) [reply src/senselab/audio/tasks/bioacoustic_qc/checks.py:26]: Good point--plan to work on here: https://github.com/sensein/senselab/issues/299
